pkg/controller/constraint/constraint_controller.go

```diff
@@ -194,9 +194,9 @@ type ReconcileConstraint struct {
 	constraintsCache *ConstraintsCache
 	tracker          *readiness.Tracker
 	getPod           func(context.Context) (*corev1.Pod, error)
-	// assumeDeleted allows us to short-circuit get requests
+	// ifWatching allows us to short-circuit get requests
 	// that would otherwise trigger a watch
-	assumeDeleted func(schema.GroupVersionKind) bool
+	ifWatching func(schema.GroupVersionKind, func() error) (bool, error)
```

Comment:
Explain the meaning of the return values. What does the "bool" signify? I understand mechanically it means "the function was executed", but I don't understand the significance.

Reply:
added

```golang
	// ifWatching allows us to short-circuit get requests
	// that would otherwise trigger a watch. The bool returns whether
	// the function was executed, which can be used to determine
	// whether the reconciler should infer the object has been deleted
```